logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/ephemeral/DeleteEphemeralMessageEndDateUseCase.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.feature.message.ephemeral
+
+/**
+ * This use case deletes ephemeral [Message] that have an end date set and it has passed current time
+ */
+interface DeleteEphemeralMessageEndDateUseCase {
```

Comment:
```suggestion
interface DeleteEphemeralMessagesAfterEndDateUseCase {
```

Suggestion: maybe a knit pick, but I think the plural makes a difference to what is doing behind the scenes.